### PR TITLE
Fix TerminationStatus

### PR DESCRIPTION
**Pull request by blegat**
Closes https://github.com/jump-dev/ECOS.jl/issues/113

### CODE REVIEW

**Comment by mtanneau** [comment]
@blegat you were faster than me :)

If it's of interest, I was writing a test for that too:
(I guess it's easier if I just post it here directly)
```julia
@testset "Iteration Limit" begin
        # Problem data
        v = [5.0, 3.0, 1.0]
        w = [2.0, 1.5, 0.3]
    
        MOI.empty!(bridged)
    
        MOI.set(bridged, MOI.RawParameter("maxit"), 1)
        MOI.set(bridged, MOI.Silent(), true)
    
        x = MOI.add_variables(bridged, 3)
        for xj in x
            MOI.add_constraint(bridged, MOI.SingleVariable(xj), MOI.Interval(0.0, 1.0))
        end
    
        # Constraint
        MOI.add_constraint(bridged, MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(w, x), 0.0), MOI.LessThan(3.0))
    
        # Set the objective
        MOI.set(bridged,
            MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(v, x), 0.0)
        )
        MOI.set(bridged, MOI.ObjectiveSense(), MOI.MAX_SENSE)
    
        MOI.optimize!(bridged)
    
        @test MOI.get(bridged, MOI.TerminationStatus()) == MOI.ITERATION_LIMIT
end
```

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -16,7 +16,8 @@ struct Solution
     objective_constant::Float64
     solve_time::Float64
 end
-const OPTIMIZE_NOT_CALLED = -1
+# The values used by ECOS are from -7 to 10 so -10 should be safe
+const OPTIMIZE_NOT_CALLED = -10
```
Comment (oxinabox) [inline src/MOI_wrapper.jl:20]: Maybe a large random negitive number, like `-82381357`, would be safer, incase they expand?

Comment (blegat) [reply src/MOI_wrapper.jl:20]: Or `nothing` might be even better

Comment (mtanneau) [reply src/MOI_wrapper.jl:20]: That would raise an error at this line: https://github.com/jump-dev/ECOS.jl/blob/1e8e25c24c367d18f6ca184fe5046618767ab9f3/src/MOI_wrapper.jl#L21

(`nothing` cannot be converted to `Int`).

Any large negative integer should do the trick.

Comment (blegat) [reply src/MOI_wrapper.jl]: I have changed the struct to allow `nothing`, I don't feel safe using magic numbers ^^

**Review by oxinabox — APPROVED** [review]

**Pull request merged**